cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -0,0 +1,2515 @@
+//! This module defines aarch64-specific machine instruction types.
+
+#![allow(non_snake_case)]
+#![allow(unused_imports)]
+#![allow(non_camel_case_types)]
+#![allow(dead_code)]
+
+use crate::binemit::CodeOffset;
+use crate::ir::constant::{ConstantData, ConstantOffset};
+use crate::ir::types::{
+    B1, B128, B16, B32, B64, B8, F32, F64, FFLAGS, I128, I16, I32, I64, I8, IFLAGS,
+};
+use crate::ir::{ExternalName, GlobalValue, JumpTable, Opcode, SourceLoc, TrapCode, Type};
+use crate::machinst::*;
+
+use regalloc::Map as RegallocMap;
+use regalloc::{
+    RealReg, RealRegUniverse, Reg, RegClass, RegClassInfo, SpillSlot, VirtualReg, Writable,
+    NUM_REG_CLASSES,
+};
+use regalloc::{RegUsageCollector, Set};
+
+use alloc::vec::Vec;
+use smallvec::{smallvec, SmallVec};
+use std::mem;
+use std::string::{String, ToString};
+
+pub mod regs;
+pub use self::regs::*;
+pub mod imms;
+pub use self::imms::*;
+pub mod args;
+pub use self::args::*;
+pub mod emit;
+pub use self::emit::*;
+
+//=============================================================================
+// Instructions (top level): definition
+
+/// An ALU operation. This can be paired with several instruction formats
+/// below (see `Inst`) in any combination.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ALUOp {
+    Add32,
+    Add64,
+    Sub32,
+    Sub64,
+    Orr32,
+    Orr64,
+    OrrNot32,
+    OrrNot64,
+    And32,
+    And64,
+    AndNot32,
+    AndNot64,
+    Eor32,
```

Comment:
Yep, it's an ARM-ism. In general we tried to maintain consistent naming w.r.t. the instruction manual unless there was a good reason otherwise. Added a doc comment to clarify.